src/inline_lazy.rs

```diff
@@ -40,6 +43,18 @@ unsafe impl<T: Sync> Sync for Lazy<T> {}
 #[doc(hidden)]
 macro_rules! __lazy_static_create {
     ($NAME:ident, $T:ty) => {
-        static mut $NAME: $crate::lazy::Lazy<$T> = $crate::lazy::Lazy(None, $crate::lazy::ONCE_INIT);
-    }
+        static mut $NAME: $crate::lazy::Lazy<$T> = $crate::lazy::Lazy::INIT;
+    };
+}
+
+#[cfg(lazy_static_core_unreachable_unchecked)]
+use core::hint::unreachable_unchecked;
+
+#[cfg(not(lazy_static_core_unreachable_unchecked))]
+/// Polyfill for core::hint::unreachable_unchecked. Included to support Rust prior to 1.27. See
+/// [issue #102](https://github.com/rust-lang-nursery/lazy-static.rs/issues/102#issuecomment-400959779)
+/// for details.
+unsafe fn unreachable_unchecked() -> ! {
```

Comment:
You don't need to do this yourself, turns out, https://docs.rs/unreachable exists.

Reply:
Hm, I think it's reasonable in this case for `lazy_static` to include this very simple function to avoid bringing in 2 additional dependencies.

Reply:
Ah that's fine, if the rationalization is added to the comment.